core/src/main/java/org/apache/carbondata/core/scan/expression/conditional/ImplicitExpression.java

```diff
@@ -59,13 +60,22 @@ public ImplicitExpression(Map<String, Set<Integer>> blockIdToBlockletIdMapping)
 
   private void addBlockEntry(String blockletPath) {
     String blockId =
-        blockletPath.substring(0, blockletPath.lastIndexOf(CarbonCommonConstants.FILE_SEPARATOR));
+        blockletPath.substring(0, blockletPath.lastIndexOf(File.separator));
+    // Check if blockletPath contains old tuple id format, and convert it to compatible format.
+    if (blockId.contains("batchno")) {
```

Comment:
Please put the strings in the constant file and then use it.

Reply:
Done